New P2P code (!!DO NOT MERGE!!)

leahcornelius opened the pull request:
This will be merged once the new p2p code is done. Please do not audit or edit the current p2p code as it will be overwrite! DO NOT MERGE THIS YET!!

leahcornelius committed:
[P2P] Begin rework of p2p

leahcornelius committed:
[P2P] Use match

leahcornelius committed:
[P2P] Begin new p2p code

leahcornelius committed:
[P2P] Update message types

leahcornelius committed:
[P2P] Begin new p2p

leahcornelius committed:
[NODE] Update refrences to P2P

leahcornelius committed:
[P2P][PEER] locking logic

leahcornelius committed:
[P2P][PEER] add_peer

leahcornelius committed:
[P2P] Use string (a ip) to index PEERS, publicise vars

leahcornelius committed:
[NODE] Create mempool

leahcornelius committed:
[NODE] Finish mempool code

leahcornelius committed:
[NODE] Fix typo

leahcornelius committed:
[RPC][NODE] Update refrences to p2p code

leahcornelius committed:
[P2P] Move to a new rx/tx based peer locking system

leahcornelius committed:
[P2P] Add new packages for encrypted messages

leahcornelius committed:
[P2P] Add new libs

leahcornelius committed:
[P2P] Add encrypted messages and a authentication tag

leahcornelius committed:
[P2P] Shell function

leahcornelius committed:
[BIN] Update refrence to p2p code

leahcornelius committed:
[NODE] Add new packeges for mempool & export it

leahcornelius committed:
[P2P] Export mempool

leahcornelius committed:
[DATABASE] Formating

leahcornelius committed:
[P2P] Add new ecnrypted 4 way handshake

leahcornelius committed:
[P2P] Typo

leahcornelius committed:
[P2P] Add peer handler code (start)

leahcornelius committed:
[P2P] Use a lazy_static for the message types hashmap, add the 4 part handhake message types

TheDevMinerTV committed:
[P2P][MessageTypes] Fix spelling and expand Ack words

leahcornelius committed:
[P2P] Recieving server code

leahcornelius committed:
[P2P] Add peer INVALID BLOCK data

leahcornelius committed:
[BLOCKCHAIN] add is defalt(&self) fn for block

leahcornelius committed:
[P2P] Clairfy message sypes expanded desc.

leahcornelius committed:
[P2P] Use a static key for sending the key derivitation result

leahcornelius committed:
[P2P] Add log P2PData function (thanks devminer!)

leahcornelius committed:
[P2P] Begin peer handler loop code

leahcornelius committed:
[P2P] Add send_block_with_hash helper function

leahcornelius committed:
[P2P][IO]Add "peek" fn

leahcornelius committed:
[P2P] Add server  code to lib exports

leahcornelius committed:
[P2P] Fix merge conflicts

leahcornelius committed:
[P2P][CARGO] Add serde_json depenedency to cargo.toml

leocornelius commented:
Ill makr as ready for review once ready and then wait on the CI/CD checks and @TheDevMinerTV's review  before merging. Not too far before its ready for merging. Maybe 1-3 days?

TheDevMinerTV commented:
Aye

leahcornelius committed:
[BIN] Update refrences to new code

leahcornelius committed:
[P2P] Add list of the peers syncing to static ref

leahcornelius committed:
[P2P] Add shell functionas

leahcornelius committed:
[P2P] Add sync close option

leahcornelius committed:
[MULTI] Cargo fix

leahcornelius committed:
[MULTI] Cargo fmt

leahcornelius committed:
[BLOCKCHAIN] Add block type to block struct

leahcornelius committed:
[P2P] Wrrite launch server wrapper code

leahcornelius committed:
[P2P] != -> ==

leahcornelius committed:
[P2P] P2P server logging

leahcornelius committed:
[P2P] debug level logging fr new incoming dtreams

leahcornelius committed:
[BLOCKCHAIN][BLOCK] Only recieve blocks can be enacted, return a error if we try to enact a send block

leahcornelius committed:
[BLOCKCHAIN] Check if block is already in mempool before enacting

leahcornelius committed:
[BLOCKCHAIN][CARGO] Change my psudoacc name to my real name

leahcornelius committed:
[BLOCKCHAIN] Dont return err if block is in mempool 

My mistake, a block will ALWAYS be in the mempool before enacting

leahcornelius committed:
[BIN] [DAEMON] Move to work with the new block format (AIP #15)

leahcornelius committed:
[BLOCKCHAIN] [GENESIS] Update genesis to work with new block tags

leahcornelius committed:
[BLOCKCHAIN] add form_recieve_block and impl default for the BlockType enum

leahcornelius committed:
[BLOCKCHAIN] Fix the dag structure (#16)

* [BLOCKCHAIN] Add block type to block struct

* [BLOCKCHAIN][BLOCK] Only recieve blocks can be enacted, return a error if we try to enact a send block

* [BLOCKCHAIN] Check if block is already in mempool before enacting

* [BLOCKCHAIN][CARGO] Change my psudoacc name to my real name

* [BLOCKCHAIN] Dont return err if block is in mempool 

My mistake, a block will ALWAYS be in the mempool before enacting

* [BIN] [DAEMON] Move to work with the new block format (AIP #15)

* [BLOCKCHAIN] [GENESIS] Update genesis to work with new block tags

* [BLOCKCHAIN] add form_recieve_block and impl default for the BlockType enum

leahcornelius committed:
[MULTI] fix issues

leahcornelius committed:
[P2P] Increase length of handshake key, aes_gcm now need at least 16 bytes

leahcornelius committed:
[P2P] Nonce size needs to be 12 now...

leahcornelius committed:
[P2P] Fix the new p2p code and the handshaking system

leahcornelius committed:
Merge branch 'fix-the-dag' into p2p-refactor

leahcornelius committed:
[MULTI] I really messed up branches here :/

* [P2P] Wrrite launch server wrapper code

* [P2P] != -> ==

* [P2P] P2P server logging

* [P2P] debug level logging fr new incoming dtreams

* [BLOCKCHAIN] Fix the dag structure (#16)

* [BLOCKCHAIN] Add block type to block struct

* [BLOCKCHAIN][BLOCK] Only recieve blocks can be enacted, return a error if we try to enact a send block

* [BLOCKCHAIN] Check if block is already in mempool before enacting

* [BLOCKCHAIN][CARGO] Change my psudoacc name to my real name

* [BLOCKCHAIN] Dont return err if block is in mempool 

My mistake, a block will ALWAYS be in the mempool before enacting

* [BIN] [DAEMON] Move to work with the new block format (AIP #15)

* [BLOCKCHAIN] [GENESIS] Update genesis to work with new block tags

* [BLOCKCHAIN] add form_recieve_block and impl default for the BlockType enum

* [MULTI] fix issues

* [P2P] Increase length of handshake key, aes_gcm now need at least 16 bytes

* [P2P] Nonce size needs to be 12 now...

* [P2P] Fix the new p2p code and the handshaking system

leahcornelius committed:
[BLOCKCHAIN] Fix the major issues with the new "dag" structure

leahcornelius committed:
[BIN] Update seednode bin, need to do this to the normal node too

TheDevMinerTV committed:
[RPC] Rewrite the API from RPC to REST

TheDevMinerTV committed:
[BIN][TESTNET][V1][DAEMON] Rework lots of the testnet v1 daemon

TheDevMinerTV committed:
Rework lots of the testnet v1 daemon

TheDevMinerTV committed:
[CONFIG] Rework a lot of the configuration

TheDevMinerTV committed:
[CONFIGURATION] Rework a lot of the configuration

TheDevMinerTV committed:
Rework a lot of the configuration

TheDevMinerTV committed:
[Database] Renaming of variables and formatting

TheDevMinerTV committed:
[Database] Renaming of variables and formatting

[DATABASE] Rework function names to remove warnings

leahcornelius committed:
[BLOCKCHAIN] Fix small bugs

leahcornelius committed:
Merge branch 'fix-the-dag' of https://github.com/avrio-project/avrio-rs into fix-the-dag

TheDevMinerTV committed:
[DATABASE] Rework function names to remove warnings

TheDevMinerTV committed:
[MULTI] Reformat and refactor

TheDevMinerTV committed:
Merge pull request #22 from avrio-project/api-rewrite

WIP: Rewrite the API

leahcornelius committed:
Merge branch 'fix-the-dag' of https://github.com/avrio-project/avrio-rs into fix-the-dag

leahcornelius committed:
[BIN] Reslove merge conflicts

leahcornelius committed:
[P2P] Port syncing code from other libary

leahcornelius committed:
[BLOCKCHAIN][BLOCK] Only add blocks related to that chain to the rec block

leahcornelius committed:
[MULTI] Add test to find max address value This will be needed for finding the range, and extrapolating using next point presision to form shards

leahcornelius committed:
[README] Add clangs dev headers (lib clang) as build requirements

As well as clang. I somehow missed that. Duh

Commit:
[NODE][VOTES] Add documentation/comments

Commit:
[BLOCKCHAIN] Minor documentation change

leahcornelius committed:
[NODE][INVITE] Fixing comments

leahcornelius committed:
[BLOCKCHAIN] Comment check block function where possable. Fixe potential preamture return issue

leahcornelius committed:
[BLOCKCHAIN] Close delimiter (trace macro)

leahcornelius committed:
[BLOCKCHAIN] Add forgoten EOS token

leahcornelius committed:
Merge branch 'fix-the-dag' into p2p-refactor

leahcornelius committed:
[MULTI] Merge p2p -> dag

* [P2P] Wrrite launch server wrapper code

* [P2P] != -> ==

* [P2P] P2P server logging

* [P2P] debug level logging fr new incoming dtreams

* [BLOCKCHAIN] Fix the dag structure (#16)

* [BLOCKCHAIN] Add block type to block struct

* [BLOCKCHAIN][BLOCK] Only recieve blocks can be enacted, return a error if we try to enact a send block

* [BLOCKCHAIN] Check if block is already in mempool before enacting

* [BLOCKCHAIN][CARGO] Change my psudoacc name to my real name

* [BLOCKCHAIN] Dont return err if block is in mempool 

My mistake, a block will ALWAYS be in the mempool before enacting

* [BIN] [DAEMON] Move to work with the new block format (AIP #15)

* [BLOCKCHAIN] [GENESIS] Update genesis to work with new block tags

* [BLOCKCHAIN] add form_recieve_block and impl default for the BlockType enum

* [MULTI] fix issues

* [P2P] Increase length of handshake key, aes_gcm now need at least 16 bytes

* [P2P] Nonce size needs to be 12 now...

* [P2P] Fix the new p2p code and the handshaking system

* [README] Add clangs dev headers (lib clang) as build requirements

As well as clang. I somehow missed that. Duh

leahcornelius committed:
[BIN][SEEDNODE] Don't re-declared the immutable wall var

leahcornelius committed:
[P2P][IO] We store peers key as hex

leahcornelius committed:
[p2p][IO] fix nonce len

leahcornelius committed:
[P2P][IO] Add step by step comments for send, read functions

leahcornelius committed:
[P2P][IO] Temp fix, always flush stream

leahcornelius committed:
[P2P][HANDLE] Update chaindigest db path

leahcornelius committed:
[MULTI] Fix handle stream not pausing during sync

leahcornelius committed:
[P2P][HANDLE] Remove surplus logging

leahcornelius committed:
[P2P][HANDLE] Misseed some ... :)

leahcornelius committed:
[P2P] Implment message type 0x60 (get chain list (ask))

leahcornelius committed:
[P2P][HANDLE] Implement message type 0x45 (get block count ask)

leahcornelius committed:
[P2P][HANDLE] Implement message type 0x6f (get blocks above hash (ask))

leahcornelius committed:
[BLOCKCHAIN] Check read chain height value is not -1 before parsing into u64

leahcornelius committed:
[BLOCKCHAIN] Form_recieve block. use recipitent public key to get top hash, not the senders

leahcornelius committed:
[BLOCKCHAIN] Logging for opening raw block files

leahcornelius committed:
[P2P][HELPER] Check block type before enacting

leahcornelius committed:
[P2P][HANDLE] Check block type before enacting

leahcornelius committed:
[P2P][HELPER] Fix block prop issues

leahcornelius committed:
[BLOCKCHAIN] Update chain_digest and state_digest system

leahcornelius committed:
[CONFIG] Change default host parameter

leahcornelius committed:
create staging branch

leahcornelius committed:
Merge branch 'staging' into p2p-refactor

leocornelius commented:
Combined with #15 and moved to #26  by #27
